src/main/java/org/geysermc/discordbot/commands/moderation/ForumPostCommand.java

```diff
@@ -0,0 +1,396 @@
+            if (!isForumChannel(event)) {
+                event.reply("Command can only be used in the forum channel").queue();
+                return;
+            }
+
+            event.getChannel().asThreadChannel().getManager()
+                .setName(event.optString("title", event.getChannel().getName()))
+                .queue(
+                    unused -> event.reply("Post was renamed").queue(),
+                    error -> event.reply("Could not rename post").queue()
+                );
+        }
+    }
+
+    public static class ClosePostSubCommand extends SlashCommand {
+        public ClosePostSubCommand() {
+            this.name = "close";
+            this.help = "Close post";
+            this.userPermissions = new Permission[] { Permission.CREATE_PUBLIC_THREADS };
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            if (!isForumChannel(event)) {
+                event.reply("Command can only be used in the forum channel").queue();
+                return;
+            }
+
+            if (event.getChannel().asThreadChannel().isArchived()) {
+                event.reply("Post is already closed").queue();
+                return;
+            }
+
+            ThreadChannelManager manager = event.getChannel().asThreadChannel().getManager();
+            event.reply("Closing post...").queue(
+                    reply -> manager.setArchived(true).queue(
+                            unused -> reply.editOriginal("Post is closed").queue(),
+                            error -> reply.editOriginal("Could not close post").queue()
+                    ));
+        }
+    }
+
+    public static class AddPostTagSubCommand extends SlashCommand {
+        public AddPostTagSubCommand() {
+            this.name = "add-tag";
+            this.help = "Add a tag to a post";
+            this.userPermissions = new Permission[] { Permission.CREATE_PUBLIC_THREADS };
+            this.options = Collections.singletonList(
+                new OptionData(OptionType.STRING, "tag", "The tag to add", true)
+                    .setAutoComplete(true)
+            );
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            if (!isForumChannel(event)) {
+                event.reply("Command can only be used in the forum channel").queue();
+                return;
+            }
+
+            ThreadChannel threadChannel = event.getChannel().asThreadChannel();
+
+            // Find the tag from the users input
+            String tag = event.optString("tag");
+            ForumTag tagFound = null;
+            for (ForumTag forumTag : threadChannel.getParentChannel().asForumChannel().getAvailableTags()) {
+                if (forumTag.getName().equalsIgnoreCase(tag)) {
+                    tagFound = forumTag;
+                    break;
+                }
+            }
+
+            if (tagFound != null) {
```

Comment:
Invert this if/else and make it a guard clause instead, e.g.

```java
if (tagFound == null) {
    event.reply("No matching tag found for " + tag).queue();
    return;
}
```
then the code in this if block